agent/handlers/task_server_setup_test.go

```diff
@@ -3942,6 +3943,13 @@ func testRegisterFaultHandler(t *testing.T, tcs []networkFaultTestCase, tmdsEndp
 
 			agentState := agentV4.NewTMDSAgentState(state, statsEngine, ecsClient, clusterName, availabilityzone, vpcID, containerInstanceArn)
 			metricsFactory := mock_metrics.NewMockEntryFactory(ctrl)
+			durationMetricEntry := mock_metrics.NewMockEntry(ctrl)
```

Comment:
I assume this is no-op for ecs agent(?)

Reply:
Yep the metric factory that ECS agent is passing is a `nopEntryFactory` object

https://github.com/aws/amazon-ecs-agent/blob/38e02f00cb46925ad982ee24ac1c8fab532ea4e2/agent/handlers/task_server_setup.go#L87

Reply:
Corresponding functions -> https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/metrics/metrics.go#L70-L73